src/driver/kvm.rs

```diff
@@ -162,6 +165,41 @@ impl<T: KVMIntrospectable> Introspectable for Kvm<T> {
         Ok(())
     }
 
+    fn get_page_access(&self, paddr: u64) -> Result<u8, Box<dyn Error>> {
+        Ok(self.kvmi.get_page_access(paddr).unwrap().try_into()?)
+    }
+
+    fn set_page_access(&self, paddr: u64, access: u8) -> Result<(), Box<dyn Error>> {
+        Ok(self
+            .kvmi
+            .set_page_access(paddr, access)
+            .unwrap()
+            .try_into()?)
+    }
+
+    fn handle_pf_event(&self, paddr: u64, pf_access: u8) -> Result<(), Box<dyn Error>> {
```

Comment:
Why have you added this function to the api? This looks to me like it is specifically tailored to your use case. In libmicrovmi we aim to provide a low level interface, not to implement convenience functions to handle events in a specific way.